### PR TITLE
Add notes about XCode also being on the Apple Developer website

### PR DESCRIPTION
**Pull request by joshuataylor**
Super minor edit about XCode being on the Apple Developer website, which can be easier to download and install without needing to go through the App Store.

Also made a note about the website requiring a free account.

Release Notes:
- N/A

### CODE REVIEW

**Comment by joshuataylor** [comment]
@cla-bot check

File: docs/src/developing_zed__building_zed.md [hunk 1]
```diff
@@ -11,7 +11,9 @@ git submodule update --init --recursive
 ## Dependencies
 
 - Install [Rust](https://www.rust-lang.org/tools/install)
-- Install [Xcode](https://apps.apple.com/us/app/xcode/id497799835?mt=12) from the macOS App Store
+- Install [Xcode](https://apps.apple.com/us/app/xcode/id497799835?mt=12) from the macOS App Store, or from the [Apple Developer](https://developer.apple.com/download/all/) website. Note this requires a developer account, which is free.
```
Comment (SomeoneToIgnore) [inline docs/src/developing_zed__building_zed.md:14]:
```suggestion
- Install [Xcode](https://apps.apple.com/us/app/xcode/id497799835?mt=12) from the macOS App Store, or from the [Apple Developer](https://developer.apple.com/download/all/) website. Note this requires a developer account.
```

`free` is a loaded term, let's not use it.

**Review by SomeoneToIgnore — APPROVED** [review]
Thanks, one minor edit I'd love to see, but good otherwise.

File: docs/src/developing_zed__building_zed.md (hunk 1 above)
Comment (joshuataylor) [reply docs/src/developing_zed__building_zed.md]: Great point, thanks! Also thanks for doing that edit.

I was trying to think of a better term to use, as I didn't want readers to think that it was tied to the $99USD/year Apple Developer Program fee.

**Pull request merged**